src/utils/hooks/useProductCategories.js

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+import { API_BASE_URL } from '../constants';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useProductCategories() {
```

Comment:
this one and the other hooks that fetch from the API have duplicated code, the only thing that is changing is the parameters of the request, this could be abstracted and reduce the duplicated logic to increase the code readability and maintainability